device/alpaka/src/clusterization/measurement_sorting_algorithm.cpp

```diff
@@ -11,30 +11,44 @@
 #include <thrust/execution_policy.h>
 #include <thrust/sort.h>
 
+#include "../utils/get_queue.hpp"
+
 namespace traccc::alpaka {
 
 measurement_sorting_algorithm::measurement_sorting_algorithm(
-    vecmem::copy& copy, std::unique_ptr<const Logger> logger)
-    : messaging(std::move(logger)), m_copy{copy} {}
+    const traccc::memory_resource& mr, vecmem::copy& copy, queue& q,
+    std::unique_ptr<const Logger> logger)
+    : messaging(std::move(logger)), m_mr{mr}, m_copy{copy}, m_queue{q} {}
 
 measurement_sorting_algorithm::output_type
 measurement_sorting_algorithm::operator()(
     const measurement_collection_types::view& measurements_view) const {
 
+    // Get a convenience variable for the queue that we'll be using.
+    auto queue = details::get_queue(m_queue);
+
     // Get the number of measurements. This is necessary because the input
     // container may not be fixed sized. And we can't give invalid pointers /
     // iterators to Thrust.
     const measurement_collection_types::view::size_type n_measurements =
         m_copy.get().get_size(measurements_view);
 
     // Sort the measurements in place
-#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
-    auto thrustExecPolicy = thrust::device;
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+    auto stream = ::alpaka::getNativeHandle(queue);
+    auto execPolicy =
+        thrust::cuda::par_nosync(std::pmr::polymorphic_allocator(&(m_mr.main)))
+            .on(stream);
+#elif defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+    auto stream = ::alpaka::getNativeHandle(queue);
+    auto execPolicy = thrust::hip_rocprim::par_nosync(
+                          std::pmr::polymorphic_allocator(&(m_mr.main)))
+                          .on(stream);
 #else
-    auto thrustExecPolicy = thrust::host;
+    auto execPolicy = thrust::host;
 #endif
```

Comment:
This is not very ideal still.

I'm going to follow up this PR with a "Thrust Unification" (name to be determined...) that gets the most appropriate execution policy from `thrust` / `rocThrust` / `oneDPL`, and then calls the relevant algorithm, since we need to do that to start to push forward SYCL support.